### DISCUSSION TITLE
Vercel Analytics

### DISCUSSION THREAD

**Discussion by bgub** [facebook/docusaurus#9079] [General]
Hey! Has anybody had luck adding Vercel Analytics to their Docusaurus site? I'd like to move away from Google Analytics.

**Comment by slorber** [comment]
@nebrelbug existing discussion, maybe ask @anaclumos

https://github.com/facebook/docusaurus/discussions/8176

**Reply by anaclumos** [reply]
I actually moved to Cloudflare Pages. Cloudflare Pages Web Analytics is way easier to integrate, it's just copy-pasting some codes.

**Reply by anaclumos** [reply]
Plus, I recommend you give Simple Analytics a try if you're moving from Google Analytics.

My referral link: https://www.simpleanalytics.com/?referral=qofuf

**Reply by bgub** [reply]
@anaclumos ok, thanks! I'd love to try something like Simple Analytics, but I'm unfortunately too poor to afford anything more expensive than free :laughing:

**Comment by bgub** [comment]
@slorber according to the Vercel Docs, it seems like I just need to put the following code in the main app file:

```js
import { inject } from '@vercel/analytics';
 
inject();
```

Any idea on how I could do that?

**Reply by anaclumos** [reply]
https://github.com/facebook/docusaurus/discussions/8176#discussioncomment-3803732

Try this!

**Comment by gracefullight** [comment]
I've simplified the installation process for Docusaurus plugins due to the hassle of verifying script domains and options. Additionally, I've added Vercel/Analytics. However, I haven't yet confirmed whether it integrates smoothly post-deployment.

We can check out the code here: https://github.com/gracefullight/docusaurus-plugins/blob/main/packages/docusaurus-plugin-vercel-analytics/src/index.ts

**Comment by bgub** [comment]
@gracefullight thanks for sharing! I've installed the plugin for production. I'll let you know if there are any issues.

**Reply by gracefullight** [reply]
Appreciated it.

**Comment by MeckenS** [comment]
@anaclumos Thank you for mentioning Cloudflare Pages Web Analytics! My Docusaurus site is hosted with Azure static web apps; however, DNS is already going through Cloudflare. I didn't even have to switch over to Cloudflare Pages - enabling Cloudflare Analytics was as easy as enabling Cloudflare proxy for my Azure static web app CNAME entries in Cloudflare. Thanks again for the tip! I didn't even realize Cloudflare offered this service (for Free).

If the site is proxied through Cloudflare, there is no need to copy the JS snippet into any code - the JS Snippet will be automatically injected.
[Sites proxied through Cloudflare](https://developers.cloudflare.com/web-analytics/get-started/#sites-proxied-through-cloudflare)
